rules/otp.bzl

```diff
@@ -0,0 +1,150 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# For a description of the OTP fields, see the files in //hw/ip/otp_ctrl/data
+
+LC_STATES = ["RAW", "DEV", "PROD", "PROD_END", "RMA", "SCRAP"]
+LC_STATES.extend(["TEST_UNLOCKED{}".format(i) for i in range(6)])
```

Comment:
I was looking at [`lc_ctrl_state.hjson`](https://cs.opensource.google/opentitan/opentitan/+/master:hw/ip/lc_ctrl/data/lc_ctrl_state.hjson) (even so, I have a typo on the `TEST_UNLOCKED` parameter it looks like).
Regarding the INVALID state, it looks like from the documentation that it is defined as any state that isn't one of the already defined ones rather than any particular value. As a result, I don't think the python script handles that as an input.

All that being said, maybe the takeaway here is that we shouldn't be doing input validation at this stage? I originally included this to prevent typos, but it seems like we should just have the underlying VMEM generating script (whether that's the current Python script or the future OTT command) handle the input.

Reply:
I've added a comment that the LC states are defined in the life cycle state definition file (by default lc_ctrl_state.hjson). The attributes themselves have short docstrings that should make it clear, and the OTP partition also spells out that it's the life cycle state.